python/tk_desktop/desktop_window.py

```diff
@@ -738,31 +785,78 @@ def _get_toolkit_classic_pipeline_configurations(self, connection, project):
         :param dict project: Project entity link.
 
         :returns: List of pipeline configuration dictionaries with keys ``mac_path``, ``windows_path``,
-            ``linux_path``, ``users`` and ``name``.
+            ``linux_path``, ``users`` and ``name``
         """
-        pipeline_configurations = connection.find(
+        found_pcs = connection.find(
             "PipelineConfiguration",
             [["project", "is", project]],
-            fields=["mac_path", "windows_path", "linux_path", "users", "code"]
-        )
-        # Ideally we would filter out in the Shotgun query any entry that has a plugin id set.
-        # Unfortunately, at the time of writing plugin ids are not live on nost Shotgun instances.
-        # So instead of filtering everything in the query, we'll filter out entries locally with the
-        # following.
-        pipeline_configurations = filter(
-            lambda pc: not("sg_plugin_ids" in pc or "plugin_ids" in pc), pipeline_configurations
+            fields=["mac_path", "windows_path", "linux_path", "users", "code", "sg_plugin_ids", "plugin_ids", "project"]
         )
 
-        for pc in pipeline_configurations:
+        def is_pipeline_usable(pc):
+            """
+            Ensures that a pipeline is not zero-config based and that it is accessible for the current user.
+            """
+            # If there is a plugin id, can't use this.
+            if pc.get("sg_plugin_ids") or pc.get("plugin_ids"):
+                return False
+
+            if not ShotgunPath.from_shotgun_dict(pc):
+                log.warning("Skipping Toolkit Classic pipeline configuration id %s without any path set." % pc["id"])
+                return False
+
+            # If there are no users assigned, this pipeline is accessible from for everyone.
+            if not pc["users"]:
+                return True
+
+            # Search for ourselves.
+            for user in pc["users"]:
+                # We've found ourselves, awesome!
+                if self._current_user_id == user["id"]:
+                    return True
+
+            return False
+
+        # FIXME: We need to discuss what we do with multiple primaries with Toolkit classic. In zero config we keep
```

Comment:
cool!

classic takes precedence, so maybe it can just check if there is a classic config - in that case, that is used, if not we fall back on bootstrap logic. Apologies if i am over simplifying :)